app/src/main/java/edu/ucandroid/weathernotice/service/FirestoreService.kt

```diff
@@ -0,0 +1,28 @@
+package edu.ucandroid.weathernotice.service
+
+import android.util.Log
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.firestore.FirebaseFirestore
+import edu.ucandroid.weathernotice.dto.Reminder
+
+class FirestoreService {
+
+    private lateinit var firestore : FirebaseFirestore
+
+    fun save(reminder: Reminder, user: FirebaseUser) {
+        val document = firestore.collection("weather").document()
+        reminder.city = document.id
+        val set = document.set(reminder)
+        set.addOnSuccessListener {
+            Log.d("Firebase", "document saved")
+            /**
+            if (reminder != null && reminder.size > 0) {
+            saveReminders(reminder, user)
+            }
+             */
+        }
+        set.addOnFailureListener {
+            Log.d("Firebase", "Save Failed")
+        }
+    }
+}
```

Comment:
From my understanding of MVVM, it is better to keep these functions in a service class. This leads to cleaner architecture in the codebase.

Citation: https://www.sovereignconsult.com/blog/know-mvvm-clean-architecture-android-app/

Reply:
You correct due to time it pushed as is at that time. Thank you